numpy/lib/histograms.py

```diff
@@ -22,6 +22,13 @@
 _range = range
 
 
+def _ptp(x):
+    # By giving an `out` argument that is floating point, we avoid the problem
+    # of signed integer arrays having a peak-to-peak value that cannot be
+    # represented with the array's data type.
+    return x.ptp(out=np.array(0.0))
```

Comment:
Would the need for `_unsigned_subtract` become immediately apparent when supporting object dtypes, or could the current code lead to subtle bugs that would take future effort to detect? It seems easier to me to avoid the analysis and use `_unsigned_subtract` now.

Reply:
@eric-wieser, do you really want me to replace one line that is correct, and is simpler and faster than your recommended alternative?  Each time I come back to "just change it already", I get irked and move on to something else.  But if y'all *really* think it should be changed, I'll grudgingly do so.  (But again, YAGNI, etc.)